File: app/src/main/java/org/wen/gank/tools/GankDao.java
package org.wen.gank.tools;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;

import org.wen.gank.model.GankModel;

import java.util.List;

import io.reactivex.Flowable;

@Dao
public interface GankDao {

    @Query("select * from gank")
    Flowable<List<GankModel>> getGanks();

    @Query("select * from gank where type = :type limit 50")
    Flowable<List<GankModel>> getGanksLimit(String type);

    @Query("select * from gank where type = :type")
    Flowable<List<GankModel>> getGanksByType(String type);

    @Insert
    void insert(GankModel model);

    @Insert
    void batchinsert(List<GankModel> models);

}

File: app/src/main/java/org/wen/gank/tools/GankDao.kt
package org.wen.gank.tools

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import io.reactivex.Flowable
import org.wen.gank.model.GankModel

@Dao
interface GankDao {

  @get:Query("select * from gank")
  val ganks: Flowable<List<GankModel>>

  @Query("select * from gank where type = :type limit 50")
  fun getGanksLimit(type: String): Flowable<List<GankModel>>

  @Query("select * from gank where type = :type")
  fun getGanksByType(type: String): Flowable<List<GankModel>>

  @Insert
  fun insert(model: GankModel)

  @Insert(onConflict = OnConflictStrategy.IGNORE)
  fun batchInsert(models: List<GankModel>)

}
